pkgs/shells/zsh/zsh-history/default.nix

```diff
@@ -0,0 +1,32 @@
+{ stdenv, lib, fetchFromGitHub, buildGoPackage }:
+
+with lib;
+
+buildGoPackage rec {
+  pname = "zsh-history";
+  version = "2019-10-27";
+
+  src = fetchFromGitHub {
+    owner = "b4b4r07";
+    repo = "history";
+    rev = "527e6f51873992fbf9c1aad70aa3009a0027a8de";
+    sha256 = "12dc380zfg3b9k7rcsyzi9dxqh28c4957b3fsx1nxvqvdm3ralm2";
+  };
+
+  goDeps = ./deps.nix;
+  goPackagePath = "github.com/b4b4r07/history";
+
+  postInstall = ''
+    install -d $out/share
+    cp -r $NIX_BUILD_TOP/go/src/${goPackagePath}/misc/* $out/share
+    cp -r $out/share/zsh/completions $out/share/zsh/site-functions
```

Comment:
You can do this using the `installShellCompletion` function, documented [here](https://hydra.nixos.org/job/nixpkgs/trunk/manual/latest/download/nixpkgs/manual.html)

Upstream seems to also have some fish functions, which might be good to install.

Reply:
Didn't know that existed. Seems to be fairly new, thanks for pointing that out :+1:

Reply:
As for fish support, please see the comment below on different shell support.